dev/NumberBox/NumberBox.cpp

```diff
@@ -81,42 +92,57 @@ void NumberBox::OnApplyTemplate()
 
 void NumberBox::OnValuePropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
-    const auto oldValue = unbox_value<double>(args.OldValue());
+    // This handler may change Value; don't send extra events in that case.
+    if (!m_valueUpdating)
+    {
+        const auto oldValue = unbox_value<double>(args.OldValue());
 
-    CoerceValue();
+        auto scopeGuard = gsl::finally([this]()
+        {
+            m_valueUpdating = false;
+        });
+        m_valueUpdating = true;
 
-    auto newValue = Value();
+        CoerceValue();
 
-    if (std::isnan(newValue) && BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten)
-    {
-        // In the validation case, we don't consider NaN to be valid.
-        newValue = oldValue;
-        Value(newValue);
-    }
+        auto newValue = Value();
 
-    if (newValue != oldValue)
-    {
-        // Fire ValueChanged event
-        const auto valueChangedArgs = winrt::make_self<NumberBoxValueChangedEventArgs>(oldValue, newValue);
-        m_valueChangedEventSource(*this, *valueChangedArgs);
+        if (std::isnan(newValue) && BasicValidationMode() == winrt::NumberBoxBasicValidationMode::InvalidInputOverwritten)
+        {
+            // In the validation case, we don't consider NaN to be valid.
+            newValue = oldValue;
+            Value(newValue);
+        }
 
-        // Fire value property change for UIA
-        if (const auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*this).as<winrt::NumberBoxAutomationPeer>())
+        if (newValue != oldValue)
         {
-            winrt::get_self<NumberBoxAutomationPeer>(peer)->RaiseValueChangedEvent(oldValue, newValue);
+            // Fire ValueChanged event
+            const auto valueChangedArgs = winrt::make_self<NumberBoxValueChangedEventArgs>(oldValue, newValue);
+            m_valueChangedEventSource(*this, *valueChangedArgs);
+
+            // Fire value property change for UIA
+            if (const auto peer = winrt::FrameworkElementAutomationPeer::FromElement(*this).as<winrt::NumberBoxAutomationPeer>())
+            {
+                winrt::get_self<NumberBoxAutomationPeer>(peer)->RaiseValueChangedEvent(oldValue, newValue);
+            }
         }
-    }
 
-    UpdateTextToValue();
+        UpdateTextToValue();
+    }
 }
 
 void NumberBox::OnMinimumPropertyChanged(const winrt::DependencyPropertyChangedEventArgs& args)
 {
+    // We only coerce the maximum to be above the minimum, not the other way around.
+    // This way, as long as the user always sets the minimum first and then the maximum,
+    // they will always get what they expect (if the values are valid).
```

Comment:
I don't think this is how range base does its Coersion. I remember you had thought about making this derive from range base, did that not pan out?

Reply:
Internally I think the framework can do coercion in a way that we can't outside. If this works well enough for the scenarios we know about or care about, is it worth spending more time on it?

Reply:
I mean.. the only thing range base does differently on rs5+ is allow coercion in both directions. When the Max is set to less then the Min it reduces the Min and when the Min is set to greater than the Max it increase the Max.  This really doesn't seem hard or something we can't do.  I agree we shouldn't check versions and change behavior if that is what you are getting at.

Reply:
I was following the pre-rs5 code in rangebase, but this make more sense. Fixed.

Reply:
Looks good, although I'm still curious why inheriting from range base didn't work out.

Reply:
Thanks! I wanted to derive from RangeBase, but I have finally wised up... there is always some divergence that results in me reimplementing the base class at the 11th hour. This time, input validation would have been a problem -- RangeBase doesn't allow you to exist in an error state. Another example: in the spec PR today we're debating making Value an IReference\<double\>.